src/Symfony/Component/Console/Command/DumpCompletionCommand.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\Process;
+
+/**
+ * Dumps the completion script for the current shell.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class DumpCompletionCommand extends Command
+{
+    protected function configure()
+    {
+        $fullCommand = $_SERVER['PHP_SELF'];
+        $commandName = basename($fullCommand);
+        $fullCommand = realpath($fullCommand) ?: $fullCommand;
+
+        $this
+            ->setName('completion')
+            ->setDescription('Dump the shell completion script')
+            ->setHelp(<<<EOH
+The <info>%command.name%</info> command dumps the shell completion script required
+to use shell autocompletion.
+
+<comment>Static installation
+-------------------</comment>
+
+Dump the script to a global completion file and restart your shell:
+
+    <info>%command.full_name% bash | sudo tee /etc/bash_completion.d/${commandName}</info>
+
+Or dump the script to a local file and source it:
```

Comment:
> Would it make sense to always generate this script as part of a cache warmer?

No, because installing the completion script is system-dependent, **the same completion script is used for multiple Symfony apps** on your system. Unless the completion is bundled with the app like I've shown in the `kubectl` example, the user needs to enable it for themselves in some way. The dynamic of updating this script will not be the same as the dynamic of building the DIC.

Also, note that, if the script is indeed installed in this manner (system-wide), cache warmer will not have permissions to update it. On some systems, a user-localized completion is supported by default, here's the snippet from Fedora:

```bash
if [ "x${BASH_VERSION-}" != x -a "x${PS1-}" != x -a "x${BASH_COMPLETION_VERSINFO-}" = x ]; then

    # Check for recent enough version of bash.
    if [ "${BASH_VERSINFO[0]}" -gt 4 ] ||
        [ "${BASH_VERSINFO[0]}" -eq 4 -a "${BASH_VERSINFO[1]}" -ge 2 ]; then
        [ -r "${XDG_CONFIG_HOME:-$HOME/.config}/bash_completion" ] &&
            . "${XDG_CONFIG_HOME:-$HOME/.config}/bash_completion"
        if shopt -q progcomp && [ -r /usr/share/bash-completion/bash_completion ]; then
            # Source completion code.
            . /usr/share/bash-completion/bash_completion
        fi
    fi

fi
```

Reply:
What about generating the file during cache warming and put in a dedicated folder of the app (with gitignore) and propose a script on the side (maybe through symfony cli) that could watch current working directory changes and auto source / unsource the file ? So it stays app specific and a good DX.